src/EFCore.PG/Metadata/Internal/NpgsqlAnnotationProvider.cs

```diff
@@ -46,43 +50,50 @@ public override IEnumerable<IAnnotation> For(IProperty property)
                 }
             }
 
-            if (property.GetTsVectorConfig() is string tsVectorConfig)
+            if (column.PropertyMappings.Select(m => m.Property.GetTsVectorConfig())
+                .FirstOrDefault(c => c != null) is string tsVectorConfig)
+            {
                 yield return new Annotation(NpgsqlAnnotationNames.TsVectorConfig, tsVectorConfig);
+            }
 
-            if (property.GetTsVectorProperties() is IReadOnlyList<string> tsVectorProperties)
+            property = column.PropertyMappings.Select(m => m.Property)
+                .FirstOrDefault(p => p.GetTsVectorProperties() != null);
+            if (property != null)
             {
                 yield return new Annotation(
                     NpgsqlAnnotationNames.TsVectorProperties,
-                    tsVectorProperties
-                        .Select(p => property.DeclaringEntityType.FindProperty(p).GetColumnName())
-                        .ToArray());
+                    property.GetTsVectorProperties().Select(p2 =>
+                        property.DeclaringEntityType.FindProperty(p2).GetColumnName()).ToArray());
```

Comment:
Rewriting this with one LINQ function per line will improve readability:

```suggestion
                    property.GetTsVectorProperties()
                        .Select(p2 => property.DeclaringEntityType.FindProperty(p2).GetColumnName())
                        .ToArray());
```

Otherwise, parenthesis counting is required to understand what's happening here.

Reply:
Done.